helix-term/src/commands.rs

```diff
@@ -1900,6 +1900,38 @@ mod cmd {
         Ok(())
     }
 
+    fn vsplit(
+        cx: &mut compositor::Context,
+        args: &[&str],
+        _event: PromptEvent,
+    ) -> anyhow::Result<()> {
+        let (_, doc) = current!(cx.editor);
+        let id = doc.id();
+
+        cx.editor.switch(id, Action::VerticalSplit);
+
+        let path = args.get(0).context("")?;
+
+        let _ = cx.editor.open(path.into(), Action::Replace)?;
```

Comment:
I think it should be enough to simply call `x.editor.open(path.into(), Action::VerticalSplit)` without the switch call?

Reply:
I force pushed a commit to handle this a bit more elegantly imo, but the Action::VerticalSplit was still needed here to allow :vsplit or :hsplit without any arguments to split the current file.